lib/RAST_SDK/AnnotationUtils.pm

```diff
@@ -415,19 +415,65 @@ sub _check_NC_features {
     }
 }
 
-
 ## Mapping the contigIDs back to their original (long) names
 sub _remap_contigIDs {
-    my ($self, $contigID_hash, $genome) = @_;
+    my ($self, $contigID_hash, $gn) = @_;
+
+    return $gn unless $contigID_hash;
 
-    my $contigs = $genome->{contigs};
-    foreach my $ctg_obj (@{$contigs}) {
-        my $cid = $ctg_obj->{id};
-        $ctg_obj->{id} = $contigID_hash->{$cid};
-        $ctg_obj->{name} = $contigID_hash->{$cid};
+    if( $gn->{contig_ids} && @{$gn->{contig_ids}} > 0 ) {
+      for my $ctg_id (@{$gn->{contig_ids}}) {
+        if( $contigID_hash->{$ctg_id} ) {
+            $ctg_id = $contigID_hash->{$ctg_id};
+        }
+      }
     }
-    return $genome;
+    if( $gn->{contigs} && @{$gn->{contigs}} > 0) {
+      for my $ctg (@{$gn->{contigs}}) {
+        if( $contigID_hash->{$ctg->{id}} ) {
+            $ctg->{id} = $contigID_hash->{$ctg->{id}};
+		}
+        if( $contigID_hash->{$ctg->{name}} ) {
+            $ctg->{name} = $contigID_hash->{$ctg->{name}};
+        }
```

Comment:
same here - you can remove the extra indentation to tidy things up a bit

Reply:
sure